scripts/screeps_tencent_batch_rl_runner.py

```
@@ -1119,6 +1130,131 @@ def safe_extract_tar(tar_path: Path, extract_dir: Path) -> None:
+
+
+def safe_policy_update_artifact_path(raw: Any, label: str = "policyUpdateArtifactPath") -> Path:
+    if not isinstance(raw, str) or not raw.strip():
+        raise BatchRunError(f"remote {label} must be a non-empty string when policyUpdateIterations is positive")
+    if "\\" in raw:
+        raise BatchRunError(f"remote {label} is unsafe: {raw!r}")
+    path = Path(raw)
+    if path.is_absolute() or ".." in path.parts:
+        raise BatchRunError(f"remote {label} is unsafe: {raw!r}")
+    if len(path.parts) < 3 or path.parts[0] != "runtime-artifacts" or path.parts[1] != "rl-training":
+        raise BatchRunError(f"remote {label} is outside rl-training artifacts: {raw!r}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)